### DISCUSSION TITLE
Sound not playing online

### DISCUSSION THREAD

**Discussion by larstuff** [pygame-web/pygbag#88] [Help zone]
Hi, thanks for Your hard work.  I have tried uploading a small game to my webserver, and the game plays well and comes up really fast. But there is no sound. I tried changing from wav to ogg, but that did not work. Hope You see this soon. 
Thanks.

**Comment by pmp-p** [comment]
hi, please provide a github link or an archive to your project or get on pygame community discord and drop a zip of it on pygame-web channel so we can investigate thank you in advance.

**Reply by larstuff** [reply]
Not so familiar with github, trying to learn fast, using pycharm. Will be back soon.

**Reply by larstuff** [reply]
The sound works now, seems like there was some messup with me deleting/renaming files. As I converted my wav file and downloaded the ogg from convertio, it automatically changed the name to 'sound_4.ogg', it was supposed to be 'sound_1.ogg'. Now game does work. See it online at (careful, this does not open in new tab)
https://naturfagmoro.com/pygbag/

Sorry for the inconvinience. Still trying to learn to clone my projects to github with pycharm. So much to learn, so little time. 

Thanks a bunch for Your tremendous effort. I teach some middle school students programming, and am convinced that learning python by starting with pygame is the way to go. If it can be played online, and maybe even on cellphones / tablets, what better way could there be to inspire youngsters to learn scripted programming, not just block coding?

I'll try get my github running with pycharm, and will definitly be back to ask for help on utilizing pygbag for pygame in the future. Have a nice day.

**Reply by larstuff** [reply]
Finally got the files up on github, after using almost 3 precious hours trying to understand Pycharm2023's strange handling of VCS and git. Think Pycharm does what it's supposed to, but installed git desktop just to be sure. Anyways, here's the link https://github.com/larstuff/pygbag-test-2023/tree/pybagger

**Reply by larstuff** [reply]
Still having som issues. Sound on iPhone has so far not successufully been accomplished. Have You looked into this? I tried wav, but doesn't play. Other formats like flac, m4a, aac, mp3 are all not supported by pygbag/pygame. Three formats; wav, mp3 and flac all play okay on win 11, x64, but pygbag will only accept wav. It also takes ogg, but this cannot be played natively on iphone.

**Reply by pmp-p** [reply]
i don't have enough funding to procure myself various apple systems to debug webkit so the only actual way is to help on pygame community discord nailing the javascript engine problem of safari, which so far seems to be located in media user engagement error handling. Some people had success on ipad removing sound auto handling with `--ume_block 0` and removing startup sound and/or wait for a mouse click.

**Reply by larstuff** [reply]
Thanks, pmp-p. Seems to be hard doing this. Won't bother You anymore with this, will follow Your advice. Btw, not only safari on iPhone / IOS, but also chrome, when installed on the system. Will try connecting with those who work with this specific problem and try finding some sort of solution.
